### PR TITLE
feat: add collection level flush rate control

### PR DESCRIPTION
**Pull request by chyezh**
flush rate control at collection level to avoid generate too much segment.
0.1 qps by default.

issue: #29477

### CODE REVIEW

**Comment by chyezh** [comment]
rerun ut

**Comment by chyezh** [comment]
rerun ut

**Comment by samhuang-z** [comment]
rerun ut

File: configs/milvus.yaml [hunk 1]
```diff
@@ -610,8 +610,10 @@ quotaAndLimits:
     enabled: false
     max: -1 # qps, default no limit, rate for CreateIndex, DropIndex
   flushRate:
-    enabled: false
+    enabled: true
```
Comment (bigsheeper) [inline configs/milvus.yaml:613]: Do we need to enable flush rate limit by default?

File: internal/proxy/multi_rate_limiter.go [hunk 2]
```diff
@@ -96,21 +96,31 @@ func (m *MultiRateLimiter) Check(collectionID int64, rt internalpb.RateType, n i
 	ret := checkFunc(m.globalDDLLimiter)
 
 	// second check collection level rate limits
-	if ret == nil && !IsDDLRequest(rt) {
-		// only dml and dql have collection level rate limits
-		ret = checkFunc(m.collectionLimiters[collectionID])
-		if ret != nil {
-			m.globalDDLLimiter.cancel(rt, n)
+	// only dml, dql and flush have collection level rate limits
+	if ret == nil && len(collectionIDs) > 0 && !IsEnableCollectionLevelLimitRequest(rt) {
```
Comment (bigsheeper) [inline internal/proxy/multi_rate_limiter.go:100]: Is**Not**EnableCollectionLevelLimitRequest?

**Comment by chyezh** [comment]
rerun ut

File: pkg/util/paramtable/quota_param.go [hunk 3]
```diff
@@ -257,6 +258,25 @@ seconds, (0 ~ 65536)`,
 	}
 	p.MaxFlushRate.Init(base.mgr)
 
+	p.MaxFlushRatePerCollection = ParamItem{
+		Key:          "quotaAndLimits.flushRate.collection.max",
+		Version:      "2.3.4",
```
Comment (czs007) [inline pkg/util/paramtable/quota_param.go:263]: 2.4?

File: pkg/util/paramtable/quota_param.go [hunk 4]
```diff
@@ -257,6 +258,25 @@ seconds, (0 ~ 65536)`,
...
+			if !p.FlushLimitEnabled.GetAsBool() {
+				return max
+			}
+			// [0 ~ Inf)
+			if getAsInt(v) < 0 {
+				return max
+			}
+			return v
+		},
+		Doc:    "qps, every 10s, rate for flush at collection level",
```
Comment (czs007) [inline pkg/util/paramtable/quota_param.go:275]: The description here can be revised. If the rate is altered, the number 10 will become ambiguous in this context.

**Pull request merged**